contractcourt/chain_watcher_test.go

```diff
@@ -213,3 +223,154 @@ func TestChainWatcherRemoteUnilateralClosePendingCommit(t *testing.T) {
 		t.Fatalf("unable to find alice's commit resolution")
 	}
 }
+
+// TestChainWatcherDataLossProtect tests that if we've lost data (and are
+// behind the remote node), then we'll properly detect this case and dispatch a
+// remote force close using the obtained data loss commitment point.
+func TestChainWatcherDataLossProtect(t *testing.T) {
+	t.Parallel()
+
+	// dlpScenario is our primary quick check testing function for this
+	// test as whole. It ensures that if the remote party broadcasts a
+	// commitment that is beyond our best known commitment for them, and
+	// they don't have a pending commitment (one we sent but which hasn't
+	// been revoked), then we'll properly detect this case, and execute the
+	// DLP protocol on our end.
+	//
+	// broadcastStateNum is the number that we'll trick Alice into thinking
+	// was broadcast, while numUpdates is the actual number of updates
+	// we'll execute. Both of these will be random 8-bit values generated
+	// by testing/quick.
+	dlpScenario := func(broadcastStateNum, numUpdates uint8) bool {
+		// If the random state num is less than our current state num,
+		// then this case will pass as we only care about the DLP case.
+		if broadcastStateNum <= numUpdates {
```

Comment:
What we could instead to is to define a new `type dlp struct { broadcastStateNum, numUpdates uint8}`, then define a `Generate` method that ensures `broadcastStateNum > numUpdates`. I think that way we get more out of our limited number of test iterations :)